src/txn/txn_ckpt.c

```diff
@@ -1641,6 +1676,67 @@ __wt_checkpoint_tree_reconcile_update(WT_SESSION_IMPL *session, WT_TIME_AGGREGAT
         }
 }
 
+/*
+ * __checkpoint_save_ckptlist --
+ *     Post processing of the ckptlist to carry forward a cached list for the next checkpoint.
+ */
+static int
+__checkpoint_save_ckptlist(WT_SESSION_IMPL *session, WT_CKPT *ckptbase)
+{
+    WT_CKPT *ckpt, *ckpt_itr;
+    WT_DECL_RET;
+    size_t ckpt_name_str_max_len;
+
+    ckpt_name_str_max_len = WT_CHECKPOINT_MAX_LEN + 1;
+
+    ckpt_itr = ckptbase;
+    WT_CKPT_FOREACH (ckptbase, ckpt) {
+        /* Remove any deleted checkpoints, by shifting the array. */
+        if (F_ISSET(ckpt, WT_CKPT_DELETE)) {
+            __wt_meta_checkpoint_free(session, ckpt);
+            continue;
+        }
+
+        /* Clean up block manager information. */
+        __wt_free(session, ckpt->bpriv);
+        ckpt->bpriv = NULL;
+
+        /* Update the internal checkpoints to their full names, with the generation count suffix. */
+        if (strcmp(ckpt->name, WT_CHECKPOINT) == 0) {
+            __wt_free(session, ckpt->name);
+            WT_RET(__wt_calloc(session, 1, ckpt_name_str_max_len, &ckpt->name));
+            ret = __wt_snprintf(
+              ckpt->name, ckpt_name_str_max_len, "%s.%" PRId64, WT_CHECKPOINT, ckpt->order);
+            if (ret != 0) {
+                __wt_free(session, ckpt->name);
+                break;
+            }
```

Comment:
This seems unnecessarily limited -- why not allocate a scratch buffer, build the string in there and strndup it into the WT_CKPT structure? It's no slower and has no maximum limit.

Reply:
done, looks much cleaner now.